blockstore/splitstore/markset_badger.go

```diff
@@ -188,6 +136,140 @@ func (s *BadgerMarkSet) Has(c cid.Cid) (bool, error) {
 	}
 }
 
+func (s *BadgerMarkSet) Visit(c cid.Cid) (bool, error) {
```

Comment:
Maybe break this into multiple functions so we can defer unlocks?

Reply:
let me think if it can be somehow be rewritten while maintaining the nice optimistic concurrency properties.

Reply:
I broke it into `tryPending` and `tryDB` methods, reused by `Has` as well.
Still not deferring the unlocks, but it is much simpler now.

Reply:
It doesn't look like it needs to be broken out into multiple functions now imo, as all the logic is in the helper functions and what this method does is lock management and gluing.